.github/workflows/pull-request.yml

```diff
@@ -15,25 +15,25 @@ jobs:
         with:
           fetch-depth: 2
 
+      - name: Restore Cache
```

Comment:
Just asking...

It seems that putting this after `npm ci` is useless thus, it was moved up, no?

Reply:
Hmmm, it's because I don't know if our restore cache strategy overrides the folder or merges the two folders.

Virtually putting it after or before didn't make significant difference, around 40-60s for NPM install (which is a lot).

I genuinely feel that we have way too many packages that we don't need, and we might want to remove some.

Usually Storybook is a big monster that has a lot of dependencies.

But anyhow, as I mentioned on an open discussion, pnpm would probably be the solution for this as it's way faster.

I also wonder what really is slowing the dependency installation down, since the caches in theory should already have all the dependencies downloaded locally...